src/fides/api/ctl/database/crud.py

```diff
@@ -163,10 +163,13 @@ async def upsert_resources(
                         )
                     )
 
+                    excluded = dict(insert_stmt.excluded.items())  # type: ignore[attr-defined]
+                    excluded.pop("id", None)  # If updating, don't update the "id"
+
                     result = await session.execute(
                         insert_stmt.on_conflict_do_update(
                             index_elements=["fides_key"],
-                            set_=insert_stmt.excluded,
+                            set_=excluded,
```

Comment:
Allison ran into trying to upsert a CTL Dataset that was already linked to a DatasetConfig and ran into:

```
fides-fides-1  | 2022-12-22 17:00:45.731 [DEBUG] (crud:upsert_resources:182): (sqlalchemy.dialects.postgresql.asyncpg.IntegrityError) <class 'asyncpg.exceptions.ForeignKeyViolationError'>: update or delete on table "ctl_datasets" violates foreign key constraint "datasetconfig_ctl_dataset_id_fkey" on table "datasetconfig"
fides-fides-1  | DETAIL:  Key (id)=(ctl_8ca1f08e-53de-4954-a56a-73123145946e) is still referenced from table "datasetconfig".
```

I believe upserting is also trying to update the `id` of an existing resource, but we need this to stay consistent, especially since DatasetConfigs now have a FK to this table.